### PR TITLE
Cleaning and refactoring test utils

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Cleaning and refactoring test utils

### Tickets:
 - [CVS-111359](https://jira.devtools.intel.com/browse/CVS-111359)

### CODE REVIEW

**Review by jane-intel — APPROVED** [review]

**Comment by olpipi** [comment]
is blocked by https://github.com/intel-innersource/applications.ai.vpu-accelerators.vpux-plugin/pull/8828

File: src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp [hunk 1]
```diff
@@ -302,6 +302,39 @@ ov::Tensor create_and_fill_tensor_consistently(const ov::element::Type element_t
     return tensor;
 }
 
+template <typename T>
+void fill_data_sin_value(T* data, size_t size) {
```
Comment (vurusovs) [inline src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp:306]: is it required? I saw such in only one test and added it inplace

**Review by vurusovs — APPROVED** [review]

File: src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp (hunk 1 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp:306]: Oh. Yes, it is unused anymore

Comment (olpipi) [reply src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp]: removed

**Review by praasz — APPROVED** [review]

File: src/plugins/template/tests/functional/skip_tests_config.cpp [hunk 2]
```diff
@@ -121,6 +121,8 @@ std::vector<std::string> disabledTestPatterns() {
         R"(.*eltwiseOpType=Mod_secondaryInputType=PARAMETER_opType=VECTOR_NetType=(f16|f32).*)",
         // Interpreter backend doesn't implement evaluate method for OP Multiply (by GroupNormalizationDecomposition)
         R"(.*ReferenceGroupNormalization.*_f64*)",
+        // CVS-131733
+        R"(.*smoke_BehaviorTests/InferRequestPropertiesTest.ReusableCPUStreamsExecutor.*)",
```
Comment (ilya-lavrenov) [inline src/plugins/template/tests/functional/skip_tests_config.cpp:125]: the test body does not contains skip condition. So, I suppose current statement does not take any effect

Comment (olpipi) [reply src/plugins/template/tests/functional/skip_tests_config.cpp:125]: The skip is in InferRequestPropertiesTest::SetUp. 
But the problem is this test fails for CPU now. So probably the root cause is not in template plugin

File: src/tests/functional/plugin/shared/include/behavior/ov_infer_request/properties_tests.hpp [hunk 3]
```diff
@@ -104,6 +104,7 @@ TEST_P(InferRequestPropertiesTest, withoutExclusiveAsyncRequests) {
 }
 
 TEST_P(InferRequestPropertiesTest, ReusableCPUStreamsExecutor) {
+    ov::threading::executor_manager()->clear();
```
Comment (vurusovs) [inline src/tests/functional/plugin/shared/include/behavior/ov_infer_request/properties_tests.hpp:107]: @olpipi couold you explain why the change is required?
